src/layouts/base-layout/types.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { ReactNode } from 'react'
+import { Products as HashiCorpProduct } from '@hashicorp/platform-product-meta'
+
+interface BaseLayoutProps {
+	/**
+	 * Content to render within the layout.
+	 */
+	children: ReactNode
```

Comment:
Nit: We should move away from manually typing `children` ourselves.

React has a `PropsWithChildren` type helper which types it for us:

Here's the signature:
```ts
    type PropsWithChildren<P = unknown> = P & { children?: ReactNode | undefined };
```

---

You can use it two ways:

```ts
interface BaseLayoutProps extends PropsWithChildren {
  ...
}

// or

type MyFullyTypedProps = PropsWithChildren<BaseLayoutProps>
```

Reply:
I dig it, TIL! 💖 Updated ✅